### PR TITLE
juce: Add version 5.4.7

### LINKED ISSUE #4020: [Request] Add JUCE
JUCE is an open-source cross-platform C++ application framework used for rapidly developing high-quality desktop and mobile applications, including VST, AU (and AUv3), RTAS and AAX audio plug-ins.

https://github.com/juce-framework/JUCE/releases

### PR DESCRIPTION
**Pull request by PlaidDroid**
- Closes #4020

### CODE REVIEW

**Comment by PlaidDroid** [comment]
What went wrong? I tested it out and it worked. Help, please!
image

File: bucket/juce.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        [
+            "Projucer.exe",
+            "Projucer"
+        ]
+    ],
+    "checkver": {
+        "github": "https://github.com/juce-framework/JUCE"
+    },
+    "autoupdate": {
+        "url": "https://github.com/cmderdev/cmder/releases/download/$version/juce-$version-windows.zip"
```
Comment (mark05e) [inline bucket/juce.json:22]: url is wrong

File: bucket/juce.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
+{
+    "homepage": "https://juce.com/",
+    "description": "JUCE is an open-source cross-platform C++ application framework. The leading framework for multi-platform audio applications",
+    "version": "5.4.7",
+    "license": {
+        "identifier": "JUCE Personal,JUCE Indie,JUCE Pro,JUCE Educational,ISC,GPL-3.0-or-later",
+        "url": "https://juce.com/juce-5-license"
+    },
```
Comment (mark05e) [inline bucket/juce.json:8]: just do
```
"license":"GPL-3.0-or-later",
```

File: bucket/juce.json [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    "bin": "Projucer.exe",
+    "shortcuts": [
+        [
+            "Projucer.exe",
+            "Projucer"
+        ]
+    ],
+    "checkver": {
+        "github": "https://github.com/juce-framework/JUCE"
+    },
```
Comment (mark05e) [inline bucket/juce.json:20]: do
```
"checkver": "github",
```

File: bucket/juce.json [hunk 4]
```diff
@@ -0,0 +1,24 @@
+{
+    "homepage": "https://juce.com/",
+    "description": "JUCE is an open-source cross-platform C++ application framework. The leading framework for multi-platform audio applications",
+    "version": "5.4.7",
```
Comment (mark05e) [inline bucket/juce.json:4]: reorder it by

1. version
2. description
3. homepage

also maybe shorten the description to something along the lines of:

> open-source cross-platform C++ application and audio applications framework.

File: bucket/juce.json (hunk 1 above)
Comment (PlaidDroid) [reply bucket/juce.json:22]: Oh! I copied an example URL! Thank you! Will fix it!

File: bucket/juce.json (hunk 2 above)
Comment (PlaidDroid) [reply bucket/juce.json:8]: It actually has a multi tier license. OK I'll shorten

File: bucket/juce.json [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+        [
+            "Projucer.exe",
+            "Projucer"
+        ]
+    ],
+    "checkver": "github",
+    "autoupdate": {
+        "url": "https://github.com/juce-framework/JUCE/releases/download/$version/juce-$version-windows.zip"
+    }
+}
```
Comment (Ash258) [inline bucket/juce.json:22]:
```suggestion
}

```

File: bucket/juce.json [hunk 6]
```diff
@@ -0,0 +1,22 @@
+{
+    "version": "5.4.7",
+    "description": "Open-source cross-platform C++ audio application framework",
+    "homepage": "https://juce.com/",
```
Comment (Ash258) [inline bucket/juce.json:4]:
```suggestion
    "homepage": "https://juce.com",
```

File: bucket/juce.json [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+    "license": {
+        "identifier": "ISC,GPL-3.0-or-later",
+        "url": "https://juce.com/juce-5-license"
+    },    
+    "url": "https://github.com/juce-framework/JUCE/releases/download/5.4.7/juce-5.4.7-windows.zip",
```
Comment (Ash258) [inline bucket/juce.json:9]:
```suggestion
    "url": "https://github.com/juce-framework/JUCE/releases/download/5.4.7/juce-5.4.7-windows.zip",
    "hash": "de88d94c0382d4ce30936666aabe4cbae0061843d28ad21130aa927fd1a7b004",
```

File: bucket/juce.json [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+    "extract_dir": "JUCE",
+    "bin": "Projucer.exe",
+    "shortcuts": [
+        [
+            "Projucer.exe",
+            "Projucer"
+        ]
+    ],
+    "checkver": "github",
```
Comment (Ash258) [inline bucket/juce.json:18]: Homepage is not github repository ....

```suggestion
    "checkver": {
        "github": "https://github.com/juce-framework/JUCE"
    },
```

File: bucket/juce.json [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+    "version": "5.4.7",
+    "description": "Open-source cross-platform C++ audio application framework",
+    "homepage": "https://juce.com/",
+    "license": {
+        "identifier": "ISC,GPL-3.0-or-later",
+        "url": "https://juce.com/juce-5-license"
+    },    
+    "url": "https://github.com/juce-framework/JUCE/releases/download/5.4.7/juce-5.4.7-windows.zip",
+    "extract_dir": "JUCE",
+    "bin": "Projucer.exe",
```
Comment (Ash258) [inline bucket/juce.json:11]: Are there any command line arguments available?

Comment (mark05e) [reply bucket/juce.json:11]: Looks like there is -- https://juce.com/discover/stories/projucer-manual#10.4-command-line-tools

**Comment by PlaidDroid** [comment]
> I do not think this should be named juice.
> 
> For scoop end user (especially in extras) it is not juce library which is needed, but Projucer application.
> 
> Description should be adjusted.

JUCE is the what people search for, that's why

File: bucket/juce.json [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+    "version": "5.4.7",
+    "description": "Open-source cross-platform C++ audio application framework",
+    "homepage": "https://juce.com",
+    "license": {
+        "identifier": "ISC,GPL-3.0-or-later",
+        "url": "https://juce.com/juce-5-license"
+    },    
+    "url": "https://github.com/juce-framework/JUCE/releases/download/5.4.7/juce-5.4.7-windows.zip",
+    "hash": "de88d94c0382d4ce30936666aabe4cbae0061843d28ad21130aa927fd1a7b004",
+    "hash": "de88d94c0382d4ce30936666aabe4cbae0061843d28ad21130aa927fd1a7b004",
```
Comment (Ash258) [inline bucket/juce.json:11]:
```suggestion
```

File: bucket/juce.json [hunk 11]
```diff
@@ -0,0 +1,25 @@
+{
+    "version": "5.4.7",
+    "description": "Open-source cross-platform C++ audio application framework",
+    "homepage": "https://juce.com",
+    "license": {
+        "identifier": "ISC,GPL-3.0-or-later",
```
Comment (Ash258) [inline bucket/juce.json:6]: There is so much information, which this list do not cover

```suggestion
        "identifier": "Proprietary",
```

File: bucket/juce.json [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+        "url": "https://juce.com/juce-5-license"
+    },    
+    "url": "https://github.com/juce-framework/JUCE/releases/download/5.4.7/juce-5.4.7-windows.zip",
+    "hash": "de88d94c0382d4ce30936666aabe4cbae0061843d28ad21130aa927fd1a7b004",
```
Comment (Ash258) [inline bucket/juce.json:10]:
```suggestion
    "architecture": {
        "64bit": {
            "url": "https://github.com/juce-framework/JUCE/releases/download/5.4.7/juce-5.4.7-windows.zip",
            "hash": "de88d94c0382d4ce30936666aabe4cbae0061843d28ad21130aa927fd1a7b004"
        }
    },
```

image

File: bucket/juce.json [hunk 13]
```diff
@@ -0,0 +1,25 @@
...
+        [
+            "Projucer.exe",
+            "Projucer"
+        ]
+    ],
+    "checkver": {
+        "github": "https://github.com/juce-framework/JUCE"
+    },
+    "autoupdate": {
+        "url": "https://github.com/juce-framework/JUCE/releases/download/$version/juce-$version-windows.zip"
```
Comment (Ash258) [inline bucket/juce.json:23]:
```suggestion
        "architecture": {
            "64bit": {
                "url": "https://github.com/juce-framework/JUCE/releases/download/$version/juce-$version-windows.zip"
            }
        }
```

**Pull request merged**